src/kubeconfig.rs

```diff
@@ -72,8 +72,10 @@ pub struct Installed {
 }
 
 impl KubeConfig {
-    pub fn write_to<W: Write>(&self, writer: W) -> anyhow::Result<()> {
-        serde_yaml::to_writer(writer, self)?;
+    pub fn write_to_file(&self, path: &Path) -> anyhow::Result<()> {
+        let file = File::create(path).context("could not write file")?;
+        fs::set_permissions(&path, Permissions::from_mode(0o600))?;
+        serde_yaml::to_writer(file, self)?;
```

Comment:
LGTM overall. Only concern is that there's no buffering on top of the `File`. `serde` makes a bunch of `write` calls with small buffers that will all end up being syscalls if no buffering is put in place.

Reply:
I think I've addressed this comment by using a `BufWriter` here... let me know please if I misunderstood. :)

Reply:
Yep that's exactly what I had in mind, thanks.